services/auth/sspi_windows.go

```diff
@@ -178,6 +178,14 @@ func (s *SSPI) shouldAuthenticate(req *http.Request) (shouldAuth bool) {
 		}
 	} else if middleware.IsAPIPath(req) || isAttachmentDownload(req) {
 		shouldAuth = true
+	} else if isGitRawReleaseOrLFSPath(req) {
+		// Allow login via SSPI if access to the repo is requested and
+		// a negotiation authorization header is present
+		_, err := sspiAuth.GetAuthData(req, nil)
+		if err == nil {
+			log.Trace("SSPI Authorization: Got authorization header, try to login")
+			shouldAuth = true
+		}
```

Comment:
Is it necessary to check if it has the authorization data that is required? The function that uses this function `func (s *SSPI) Verify` is already checking this.

Reply:
Yes, as otherwise the `Verify` function sends a 401/StatusUnauthorized response and requests authentication even for requests, that need no authentication (eq public repos). 
Adding this check here only performs the verification if a `Autorization: Negotiation ..."` header is present and thus has been requested from the `http_base`  function or if the client directly sends this header.

Maybe if the `Verify` function is changed, so that the 401 response is not directly sent on authentication failure this check is not needd anymore.

Reply:
I have question, why only `isGitRawReleaseOrLFSPath` paths are checked? What about other paths? IMO either all paths (for Web UI) should be checked together, or none of them should be checked and leave the work to other modules. Or there should be a comment about why only `isGitRawReleaseOrLFSPath` paths are checked.

Reply:
good point, the intention was to not change the behavoir of the other requests.
I now just did a quick try on the API and (probably) found a bug: Anonymous API access is not possible if SSPI is enabled as authsource. Accessing via swagger leads to the login page and access via curl returns the login page as HTML instead of a JSON response.